lib/ansible/plugins/inventory/onesixtyone.py

```diff
@@ -0,0 +1,244 @@
+      - 'TODO: Add caching support.'
+'''
+EXAMPLES = '''
+    # onesixtyone.yaml or any other YAML file-extension
+    plugin: onesixtyone
+    strict: False
+    addresses:
+      - 192.168.0.0/24
+'''
+
+import os
+import re
+import tempfile
+import subprocess
+
+from ansible import constants as C
+from ansible.errors import AnsibleParserError
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common.process import get_bin_path
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+try:
+    from netaddr import IPSet
+    HAS_NETADDR = True
+except Exception:
+    HAS_NETADDR = False
+
+try:
+    import gevent
+    from gevent import socket
+    HAS_GEVENT = True
+except Exception:
+    HAS_GEVENT = False
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    NAME = 'onesixtyone'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+        """ Verify the existence of plugin's configuration file """
+        if not super(InventoryModule, self).verify_file(path):
+            return False
+
+        for suffix in ('yaml', 'yml'):
+            maybe = '{fn}.{suffix}'.format(fn=self.NAME, suffix=suffix)
+            if path.endswith(maybe):
+                return True
+
+    def parse(self, inventory, loader, path, cache=False):
+        super(InventoryModule, self).parse(inventory, loader, path, cache=cache)
+        self._onesixtyone = get_bin_path(self.NAME, True)
+        self._read_config_data(path)
+
+        if not HAS_NETADDR:
+            raise AnsibleParserError('this inventory plugin requires the netaddr library. Try: pip install netaddr')
+
+        if not HAS_GEVENT:
+            raise AnsibleParserError('this inventory plugin requires the gevent library. Try: yum install python-gevent')
+
+        try:
+            # Using gevent to parallelize reverse dns lookups and adding entries to inventory
+            jobs = [gevent.spawn(self._add_device, self, device) for device in self._discovery()]
+            gevent.joinall(jobs, timeout=20)
+
+        except Exception as e:
+            raise AnsibleParserError("failed to parse %s: %s " % (to_native(path), to_native(e)))
+
+    def _match_platform(self, text):
+        """ Matches platforms against text in sysDescr and return the platform.
+
+        Keyword arguments:
+        text -- sysdescr text
+        """
+        platforms = {
+            'asa': re.compile(r'^Cisco\sAdaptive\sSecurity\sAppliance'),
+            'ios': re.compile(r'^Cisco\sIOS\sSoftware'),
+            'iosxr': re.compile(r'^Cisco\sIOS\sXR\sSoftware'),
+            'nxos': re.compile(r'^Cisco\sNexus\sOperating\sSystem'),
+            'dellos': re.compile(r'^Dell\sApplication\sSoftware'),
+            'junos': re.compile(r'^JUNOS'),
+            'aruba': re.compile(r'^Aruba\sOperating\sSystem\sSoftware'),
+            'eos': re.compile(r'^Arista')
```

Comment:
I prefer to have a trailing comma everywhere, it makes further diff review easier.
```suggestion
            'eos': re.compile(r'^Arista'),
```

Reply:
Added ','